### DISCUSSION TITLE
How to use invalidate queries to refetch queries.

### DISCUSSION THREAD

**Discussion by yourcasualdev** [wevm/wagmi#2550] [Q&A]
I asked this question on [Stackoverflow](https://stackoverflow.com/questions/76482333/how-to-refech-queries-in-wagmi) but didn't get any answer so far so I came here. I want to use invalidate queries to refetch Wagmi's query functions. After all, it uses react-query under the hood. So what is the query key on below function:
```typescript
  const balance = useContractRead({
    address: --random-adress--,
    abi: random.abi,
    functionName: 'balanceOf',
    onSuccess: () => {
      console.log('balanceOf refeched');
    },
    cacheTime: 0,
    staleTime: 0,
    scopeKey: 'balanceOf',
  });
```

functionName, scopeKey, or something else?

**Comment by WingsDevelopment** [comment]
any progress on this?

**Comment by alex-pominov** [comment]
I found out that you can invalidate/refetch queries passing just a simple string instead of array to `resetQueries`/`refetchQueries` methods of `queryClient`.

```ts
import { useQueryClient } from "wagmi";

const queryClient = useQueryClient();
// Invalidate/refetch a specific query by its key
queryClient.resetQueries({ queryKey: "balanceOf" as never });
```

Please note, it's important to correctly set the `scopeKey` for your queries to ensure they are accurately targeted for invalidation or refetching.

**Comment by 0x33dm** [comment]
>  I want to use invalidate queries to refetch Wagmi's query functions. After all, it uses react-query under the hood. So what is the query key on below function:

Have a look [at the documentation](https://wagmi.sh/react/guides/tanstack-query#query-keys) `useReadContract` will return a `queryKey` !

I hope that helps
